src/Kiota.Builder/Writers/CSharp/CodeEnumWriter.cs

```diff
@@ -25,13 +25,8 @@ public override void WriteCodeElement(CodeEnum codeElement, LanguageWriter write
                             .Select(x => x.ToFirstCharacterUpperCase())
                             .Select((x, idx) => $"{x}{(codeElement.Flags ? " = " + GetEnumFlag(idx) : string.Empty)},")
                             .ToArray());
-            writer.DecreaseIndent();
-            writer.WriteLine("}");
-            if(codeNamespace != null) {
-                writer.DecreaseIndent();
-                writer.WriteLine("}");
-            }
-            
+            if(codeNamespace != null)
```

Comment:
I believe we are missing a before this if statement
```cs
writer.CloseBlock();
```

Reply:
no, thanks to the refactoring I've made, code enum is now using a BlockEnd for its "End" property. And this gets picked up by the BlockEndWriter (formerly CodeClassEndWriter), hence why I cleaned it up at multiple places.

Reply:
Sorry, I got it now. The `CodeBlockEndWriter` will kick in and close the block. Yes?